lib/seoshop-api/client/order.rb

```diff
@@ -0,0 +1,62 @@
+module Seoshop
+  class Client::Order
+    class CheckoutError
+    end
+
+    attr_reader :id, :client, :language, :checkout_details, :payment_details, :shipping_details, :checkout_id, :checkout_body
+    alias :before_validation :update_shipping_and_paymeny_info
+
+    def initialize(client, checkout_details, payment_details, shipping_details)
+      @client = client
+      @language = language
+      @payment_details = payment_details
+      @shipping_details = shipping_details
+      @checkout_response = @client.post("#{language}/checkouts.json", checkout_details)
+      @checkout_id ||= @checkout_response[:id]
+      @checkout_body ||= @checkout_response.body
+    end
+
+    def valid?
+      before_validation
+      client.get("#{language}/checkouts/#{checkout_id}/validate.json").body[:validated]
+    end
+
+    def save!
+      response = client.post("#{language}/checkouts/#{checkout_id}/order.json", checkout_attrs)
+      if response.status == 201 || response.status == 200
+        @id = response.body[:order_id]
+      else
+        error = response.body[:error]
+        fail CheckoutError, "Checkout with id: #{checkout_id} could not convert into an order. Errors: #{error[:message]} (code: #{error[:code]})"
+      end
+    end
+
+    def update(attrs)
+      response = client.put("#{language}/orders/#{id}.json", { order: attrs })
+      unless response.status == 200
+        fail CheckoutError, "Order with id: #{id} could not be updated."
+      end
+    end
+
+    private
+
+    def before_validation
+      update_shipping_and_paymeny_info
+    end
+
+    def update_shipping_and_paymeny_info
+      response = client.put("#{language}/checkouts/#{checkout_id}.json", { shipping_method: shipping_details, payment_method: payment_details })
+      fail CheckoutError, "Checkout with id: #{checkout_id} could not be updated." unless response.status == 200
+    end
+
+    def checkout_attrs
+       sanitize_hash checkout_body.slice(:mode, :customer, :billing_address, :shipping_address, :newsletter, :terms, :notifications).
+       merge(quote_id: checkout_body[:quote][:id]).
```

Comment:
can you indent here and maybe place the dot (`.`) on the newline instead of end of the line (rubocop styleguide)